lua/luasnip/extras/init.lua

```diff
@@ -48,12 +48,16 @@ local function to_function(val, use_re)
 	assert(false, "Can't convert argument to function")
 end
 
+local function _idem(x)
+	return x
```

Comment:
How about instead doing `return _match(x) or ""`, that would allow to return the entire matched text or the first capture group by default, which would be quite intuitive?

Reply:
(`_match` from after `to_function`)

Reply:
That will be problematic if someone does something like 
`m(1,  tonumber)`

Reply:
true, I assumed the lambdas we get as `_match` would always end with a `match()`.
Your case is problematic, but what would the expected result be when putting `tonumber` as the match?
(we could also test for type==string to protect against that)

Reply:
I used isnumber as an example, the idea is that any function could be used there and that make sence as one could want to use a custom function to check some property of the text, like being a number, or other more complex that can't be expressed in a pattern.

Reply:
Ah, yeah the function-as-pattern is very nice to have.
But even in that case, using `return (type(_match(x))=="string" and _match(x)) or ""` wouldn't cause any problems, true, the matched number would have to be inserted using `_then`(or with `m(1, "^%d+$")`), but in the vast majority(I'd think) of cases (simple regex), that would be very useful, eg `m(1, ".?.?.?")` to return up to the first three characters.

Reply:
More complex stuff is bound to be implemented by the user, where they can return a string(which will be inserted if no `_then` is given) on match or `nil`

Reply:
I'm far from home today if you wanna have it soon feel free to merge and fix it, or I'll do it later.

Reply:
Okay, sounds good:+1: